src/components/node-list/styles/_filter.scss

```diff
@@ -0,0 +1,115 @@
+@import '../../../styles/extends';
+@import '../../../styles/mixins';
+@import '../../../styles/variables';
+@import './variables';
+
+.pipeline-nodelist__item--is-filter {
+  .kui-theme--dark & .pipeline-nodelist__row__visibility-icon {
```

Comment:
I see what you mean but I'd suggest looking at the whole file as that would require a lot of duplication, the modifier would need to be added in a lot of places, which could also break separation of concerns.

The selectors would become much longer too since a lot of them are already modifiers, so you'd still end up with two classes anyway. If it's the nesting you don't like I could manually add the prefix class in everywhere, but I feel that would be very bad for readability.

Reply:
It's not the nesting that's the problem: As far as I'm concered, these are equivalent:
```scss
.parent .child {
  ...
}

.parent {
  .child {
    ...
  }
}
```

The problem is the increased CSS specificity, which makes it much harder to maintain if you want to override CSS rules later without escalating specificity. I think it's worth the longer class names - if you want, you can reduce this `pipeline-nodelist__row__visibility-icon` down to something like `pipeline-visibility-icon` if it helps.

Reply:
That works for some of the selectors in this file but not all, e.g.

`.pipeline-nodelist__item--is-filter .pipeline-nodelist__row__visibility-icon--checked`

will become:

`.pipeline-nodelist__row__visibility-icon--is-filter.pipeline-nodelist__row__visibility-icon--checked`

Which I think are the same specificity, so we'd have to add combined versions to get lower:

`.pipeline-nodelist__row__visibility-icon--is-filter--checked`

Is that what you mean?

Reply:
Sure I've applied this and simplified it a bit further. This does come at a cost of increasing coupling on the JS side, but in this case that seems natural.